### DISCUSSION TITLE
Is "channels" in INSTALLED_APPS required in production?

### DISCUSSION THREAD

**Discussion by jonaswinkler** [django/channels#1632] [Q&A] [answered]
Hello!

The documentation states that we should add "channels" to INSTALLED_APPS. This will in turn enable `runserver` to run the ASGI application. I've noticed that running my application in production with daphne or gunicorn+uvicorn works fine even without "channels" in the app list.

*Question:* Are there any drawbacks to not including 'channels' in INSTALLED_APPS? I'm considering to include the "channels" application only when DEBUG is enabled.

Background: I'm running an application that also starts a lot of background processes, each importing the settings and therefore channels (+daphne and dependencies) as well. That's quite a bit of allocated resources that aren't exactly needed, and the app is also supposed to run decent on low power devices.

Thank you very much!

**Comment by carltongibson** [comment, marked as answer]
Installed apps are needed for detection of models, commands and templates. (I think that's it). 

Likely yes, you don't need those.

**Reply by jonaswinkler** [reply]
Thanks for the clarification.

I've looked at the source before, and the only thing the `channels` application does is to monkey patch the `runserver` command. I was not entirely sure if including the `channels` app has some other required side effects that aren't immediately visible.

**Comment by KStenK** [comment]
After we had the same issues like #1639 and our production server workers did not boot up, then removing channels from INSTALLED_APPS fixed that problem for us and everything seems to work fine so far.
